### ISSUE TITLE
Declare Typescript custom typings

### ISSUE THREAD

**Issue by pedrobranco** [ruimarinho/bitcoin-core#140]

**Comment by ChandlerFerry** [comment]
@pedrobranco What strategy do you want to work towards for this?

- Convert to TypeScript?
- Declaration file in this repo with NPM "types" property?
- Declaration file in DefinitelyTyped?

While I do not plan to solve the issue, I can begin the process to make it easier for other users to continue to build upon.

**Comment by pedrobranco** [comment]
@ruimarinho WDYT of using declaration file in `DefinitelyTyped`?

**Comment by LeeKorbis** [comment]
How is anyone using TS at all in the Bitcoin ecosystem?

* bitcoinjs/coinselect - no types
* ruimarinoho/bitcoin-core - no types
* bitcoinjs/bitcoinjs-lib - types

Another option not mentioned is using JSDoc. https://www.typescriptlang.org/docs/handbook/jsdoc-supported-types.html

I'll add the JSDoc annotations since nothing exists and open a PR. I only hope it'll be accepted. :laughing: I'll base it on https://github.com/ruimarinho/bitcoin-core/pull/149 - This makes the types completely orthogonal. 

I've done the same with bitcoinjs/coinselect. I'll be opening a PR there as well.

**Comment by 0-don** [comment]
yes types would be nice, why not just add at least the types from index.d.ts in this old puill request https://github.com/ruimarinho/bitcoin-core/pull/65

**Comment by danijelTxFusion** [comment]
What's the status with this?
